File: backend/app/schemas/criterion.py
# backend/app/schemas/criterion.py
from pydantic import BaseModel, Field, validator
from uuid import UUID
from typing import Optional, List
from decimal import Decimal

class CriterionCreate(BaseModel):
    title: str = Field(..., max_length=150)
    description: Optional[str] = None
    max_score: float = Field(..., gt=0)
    weight_percent: float = Field(..., ge=0, le=100)
    sort_order: int = Field(..., gt=0)
    is_active: bool = True

    @validator('max_score')
    def validate_max_score(cls, v):
        if v <= 0:
            raise ValueError('max_score must be greater than 0')
        return v

class CriterionUpdate(BaseModel):
    title: Optional[str] = Field(None, max_length=150)
    description: Optional[str] = None
    max_score: Optional[float] = Field(None, gt=0)
    weight_percent: Optional[float] = Field(None, ge=0, le=100)
    sort_order: Optional[int] = Field(None, gt=0)
    is_active: Optional[bool] = None

class CriterionResponse(BaseModel):
    id: UUID
    title: str
    description: Optional[str] = None
    max_score: float
    weight_percent: float
    sort_order: int
    is_active: bool
    created_at: datetime
    updated_at: datetime

class CriteriaListResponse(BaseModel):
    items: List[CriterionResponse]
    total_weight: float
    weights_valid: bool

class CriteriaReorderItem(BaseModel):
    id: UUID
    sort_order: int

class CriteriaReorderRequest(BaseModel):
    items: List[CriteriaReorderItem]

File: frontend/lib/models/criterion.dart
class Criterion {
  final String id;
  final String title;
  final String? description;
  final double maxScore;
  final double weightPercent;
  final int sortOrder;
  final bool isActive;

  Criterion({
    required this.id,
    required this.title,
    this.description,
    required this.maxScore,
    required this.weightPercent,
    required this.sortOrder,
    required this.isActive,
  });

  factory Criterion.fromJson(Map<String, dynamic> json) {
    return Criterion(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      maxScore: (json['max_score'] as num).toDouble(),
      weightPercent: (json['weight_percent'] as num).toDouble(),
      sortOrder: json['sort_order'] ?? 0,
      isActive: json['is_active'] ?? true,
    );
  }
}

class CriteriaListResponse {
  final List<Criterion> items;
  final double totalWeight;
  final bool weightsValid;

  CriteriaListResponse({
    required this.items,
    required this.totalWeight,
    required this.weightsValid,
  });

  factory CriteriaListResponse.fromJson(Map<String, dynamic> json) {
    return CriteriaListResponse(
      items: (json['items'] as List?)
              ?.map((e) => Criterion.fromJson(e))
              .toList() ??
          [],
      totalWeight: (json['total_weight'] as num?)?.toDouble() ?? 0.0,
      weightsValid: json['weights_valid'] ?? false,
    );
  }
}
